src/codecs/avif/decoder.rs

```diff
@@ -86,13 +112,209 @@ impl<R: Read> AvifDecoder<R> {
+                UnsupportedErrorKind::GenericFeature(
+                    "Using 'Reserved' color matrix is not supported".to_string(),
+                ),
+            ),
+        )),
+        dav1d::pixel::MatrixCoefficients::BT470M => Ok(YuvStandardMatrix::Bt470_6),
+        dav1d::pixel::MatrixCoefficients::BT470BG => Ok(YuvStandardMatrix::Bt601),
+        dav1d::pixel::MatrixCoefficients::ST170M => Ok(YuvStandardMatrix::Smpte240),
+        dav1d::pixel::MatrixCoefficients::ST240M => Ok(YuvStandardMatrix::Smpte240),
+        // This is an experimental matrix in libavif yet.
+        dav1d::pixel::MatrixCoefficients::YCgCo => Err(ImageError::Unsupported(
+            UnsupportedError::from_format_and_kind(
+                ImageFormat::Avif.into(),
+                UnsupportedErrorKind::GenericFeature("YCgCo matrix is not supported".to_string()),
+            ),
+        )),
+        dav1d::pixel::MatrixCoefficients::BT2020NonConstantLuminance => {
+            Ok(YuvStandardMatrix::Bt2020)
+        }
+        dav1d::pixel::MatrixCoefficients::BT2020ConstantLuminance => {
+            // This matrix significantly differs from others because linearize values is required
+            // to compute Y instead of Y'.
+            // Actually it is almost everywhere is not implemented.
+            // Libavif + libheif missing this also so actually AVIF images
+            // with CL BT.2020 might be made only by mistake
+            Err(ImageError::Unsupported(
+                UnsupportedError::from_format_and_kind(
+                    ImageFormat::Avif.into(),
+                    UnsupportedErrorKind::GenericFeature(
+                        "BT2020ConstantLuminance matrix is not supported".to_string(),
+                    ),
+                ),
+            ))
+        }
+        dav1d::pixel::MatrixCoefficients::ST2085 => Err(ImageError::Unsupported(
+            UnsupportedError::from_format_and_kind(
+                ImageFormat::Avif.into(),
+                UnsupportedErrorKind::GenericFeature("ST2085 matrix is not supported".to_string()),
+            ),
+        )),
+        dav1d::pixel::MatrixCoefficients::ChromaticityDerivedConstantLuminance
+        | dav1d::pixel::MatrixCoefficients::ChromaticityDerivedNonConstantLuminance => Err(
+            ImageError::Unsupported(UnsupportedError::from_format_and_kind(
+                ImageFormat::Avif.into(),
+                UnsupportedErrorKind::GenericFeature(
+                    "Chromaticity Derived Luminance matrix is not supported".to_string(),
+                ),
+            )),
+        ),
+        dav1d::pixel::MatrixCoefficients::ICtCp => Err(ImageError::Unsupported(
+            UnsupportedError::from_format_and_kind(
+                ImageFormat::Avif.into(),
+                UnsupportedErrorKind::GenericFeature(
+                    "ICtCp Derived Luminance matrix is not supported".to_string(),
+                ),
+            ),
+        )),
```

Comment:
Ah, yes, there's an additional non-linearity to consider in this case. Let's implement this is a seperate PR, should be easier with the infrastructure in place.

Reply:
As to me actual issue is not an implementation itself here, as also for CL 2020, YCgCo and others, is that there is no reference software allows you to do make those image and view them, or at least I don't know anything.

AVIF's with `Bt.2020 CL` CICP exists only in Netflix samples avif repository and [author of libavif](https://github.com/AOMediaCodec/libavif/issues/324)marked that it was highly likely made by mistake.

I never saw intentionally, and correctly encoded image with CICP have `ICtCp` matrix as well as Bt.2020 CL also.
If you have one in your possesion it is interesting to have a look on it, and probably then ask what reference software may do so.

Reply:
As for `YCgCo`, I think I can also do commit with it but after this PR successfully lands, and, perhaps, if you'll make this in separate repo in your organisation it also might be more convenient.
Even `YCgCo` might be blend in current implementation, actually, it was made to make possible encoding using only bit shift operators, and much simpler decoding process, so in terms of performance, different implementation might be preferred. 
Here is in example of `YCgCo` decoding and for limited range it is even simpler.
```rust
    let y_value = (y_plane[0] as i32 - bias_y)
                * range_reduction_y;
            
            let cg_value =
                (cg_plane[0] as i32 - bias_uv) * range_reduction_uv;

            let co_value =
                (co_plane[0] as i32 - bias_uv) * range_reduction_uv;

            let t = y_value - cg_value;

            let r = ((t + co_value + ROUNDING_CONST) >> PRECISION)
                .min(255)
                .max(0);
            let b = ((t - co_value + ROUNDING_CONST) >> PRECISION)
                .min(255)
                .max(0);
            let g = ((y_value + cg_value + ROUNDING_CONST) >> PRECISION)
                .min(255)
                .max(0);
```

Reply:
Good point on missing (public) reference software. Without any validation cases it'll indeed be quite hard to land it with confidence. This is definitely a separate issue to discuss, let's not trail too far off-topic in this PR.

Reply:
AVIF has inherited HEIF and a video codec, which have tons of features and variants, many of which exist just because an older standard had them, and are of questionable utility. 

I suggest implementing only features that are supported across web browsers. That's a de-facto practical profile of the format.

Reply:
YCgCo is a neat color space for lossless encoding, but lossless encoding in AV1 is pretty weak.

Reply:
YCgCo [has added to libavif](https://github.com/AOMediaCodec/libavif/issues/2077) so as it exists in reference implementation in some future it will be likely added everywhere ( and I it looks like it is [already added](https://source.chromium.org/chromium/chromium/src/+/main:media/base/video_color_space.h;l=69;drc=4f00803ca25c0d0480ed14844d6406933c21e80e) )
As for others, CL, chromacity derived and others, I'd agree.

If you want to target as browsers do then issue with `Unspecified` still in the air because Chromium [definitely goes always with Bt.709](https://source.chromium.org/chromium/chromium/src/+/main:media/base/video_color_space.h;l=69;drc=4f00803ca25c0d0480ed14844d6406933c21e80e).

Reply:
> YCgCo is a neat color space for lossless encoding, but lossless encoding in AV1 is pretty weak.

I think it is just a question to @HeroicKatora and @fintelia if they want to support that or not rather that encoding quality.

Reply:
And it is looks like [libavif](https://github.com/AOMediaCodec/av1-avif/issues/84) also want to consider unspecified as Bt.709.